src/gpu/d3d12/SDL_gpu_d3d12.c

```diff
@@ -2473,12 +2473,31 @@ static D3D12GraphicsRootSignature *D3D12_INTERNAL_CreateGraphicsRootSignature(
     return d3d12GraphicsRootSignature;
 }
 
+static bool D3D12_INTERNAL_IsValidShaderBytecode(
+    const Uint8 *code,
+    size_t codeSize)
+{
+    // Both DXIL and DXBC bytecode have a 4 byte header containing `DXBC`.
+    if (codeSize < 4)
+        return false;
+    return code[0] == 'D' && code[1] == 'X' && code[2] == 'B' && code[3] == 'C';
```

Comment:
Using `SDL_memcmp` would make the intention cleaner, I think.

Also applies to other instances.